libfastpm/cosmology.c

```diff
@@ -323,7 +333,21 @@ static ode_soln growth_ode_solve(double a, FastPMCosmology * c)
     return soln;
 }
 
-/* FIXME: Each of these functions will redo the ode! That's long! */
+void fastpm_growth_info_init(FastPMGrowthInfo * growth_info, double a, FastPMCosmology * c) {
+    ode_soln soln = growth_ode_solve(a, c);
+    ode_soln soln_a1 = growth_ode_solve(1, c);
+    // FIXME: you could save a=1 soln at the start of code (perhaps to cosmo obj) and then never compute again.
+
+    growth_info->a = a;
+    growth_info->c = c;
+    growth_info->D1 = soln.y0 / soln_a1.y0;
+    growth_info->f1 = soln.y1 / soln.y0;    /* f = d log D / d log a. Note soln.y1 is d d1 / d log a */
+    growth_info->D2 = soln.y2 / soln_a1.y2;
+    growth_info->f2 = soln.y3 / soln.y2;
+}
+
+/* FIXME: Some of the below growth functions are still called in io.c, horizon.c and solver.c.
+          This shouldn't be a big deal, but might want to change to growth object later. */
```

Comment:
Yes. I think it is a good idea to fully migrate to GrowthInfo. There also used to call fastpm_solver_growth_factor etc, and what really needed is a GrowthInfo interface if you haven't done that.

Reply:
Done. Note, I just removed the funcs in solver.c, and directly made growth objects in fastpm.c as this is the only place fastpm_solver_growth_... was called. Is this good?